src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -1764,11 +1764,11 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
     fn append_local_to_string(&self, local_index: Local, buf: &mut String) -> Result<(), ()> {
         let local = &self.mir.local_decls[local_index];
         match local.name {
-            Some(name) => {
-                buf.push_str(&name.to_string());
+            Some(name) if !local.from_compiler_desugaring() => {
```

Comment:
can you update the doc-comment for this method to reflect its behavior more accurately? Something like:

```
/// Appends end-user visible description of the `local` place to `buf`. If `local` doesn't have
/// a name, or if its name was generated by the compiler, then `Err` is returned and `buf` is left unmodified.
```